src/providers/hana/qgshanaprovider.cpp

```diff
@@ -0,0 +1,1798 @@
+/***************************************************************************
+   qgshanaprovider.cpp  -  Data provider for SAP HANA
+   --------------------------------------
+   Date      : 31-05-2019
+   Copyright : (C) SAP SE
+   Author    : Maxim Rylov
+ ***************************************************************************/
+
+/***************************************************************************
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ ***************************************************************************/
+#include "qgis.h"
+#include "qgsapplication.h"
+#include "qgscoordinatereferencesystem.h"
+#include "qgsdataprovider.h"
+#include "qgsfeature.h"
+#include "qgsfields.h"
+#include "qgsgeometry.h"
+#include "qgshanaconnectionpool.h"
+#include "qgshanaexception.h"
+#include "qgshanadriver.h"
+#include "qgshanafeatureiterator.h"
+#include "qgshanaprovider.h"
+#include "qgshanaproviderconnection.h"
+#include "qgshanaresultset.h"
+#include "qgshanautils.h"
+#ifdef HAVE_GUI
+#include "qgshanadataitems.h"
+#include "qgshanasourceselect.h"
+#include "qgssourceselectprovider.h"
+#endif
+#include "qgslogger.h"
+#include "qgsmessagelog.h"
+#include "qgsrectangle.h"
+
+#include "ogr_srs_api.h"
+
+#include <ctype.h>
+
+#include "odbc/PreparedStatement.h"
+#include "odbc/ResultSet.h"
+#include "odbc/ResultSetMetaDataUnicode.h"
+
+using namespace odbc;
+using namespace std;
+
+namespace
+{
+  QString buildQuery( const QString &source, const QString &columns, const QString &where, const QString &orderBy, int limit )
+  {
+    QString sql = QStringLiteral( "SELECT %1 FROM %2" ).arg( columns, source );
+    if ( !where.isEmpty() )
+      sql  += QStringLiteral( " WHERE " ) + where;
+    if ( !orderBy.isEmpty() )
+      sql  += QStringLiteral( " ORDER BY " ) + orderBy;
+    if ( limit >= 0 )
+      sql  += QStringLiteral( " LIMIT " ) + QString::number( limit );
+    return sql;
+  }
+
+  void createCoordinateSystem( QgsHanaConnectionRef &conn, const QgsCoordinateReferenceSystem &srs )
+  {
+    OGRSpatialReferenceH hCRS = OSRNewSpatialReference( nullptr );
+    int errcode = OSRImportFromProj4( hCRS, srs.toProj().toUtf8() );
+    if ( errcode != OGRERR_NONE )
+    {
+      if ( hCRS )
+        OSRRelease( hCRS );
+      throw QgsHanaException( "Unable to parse a spatial reference system" );
+    }
+
+    QgsCoordinateReferenceSystem srsWGS84;
+    srsWGS84.createFromString( "EPSG:4326" );
+
+    QgsCoordinateTransform transform;
+    transform.setSourceCrs( srsWGS84 );
+    transform.setDestinationCrs( srs );
```

Comment:
This needs to use a QgsCoordinateTransformContext -- otherwise it's calling deprecated API

Reply:
Done in 1fa8b839a6.